### PR TITLE
Enarx updates

### PR DESCRIPTION
**Pull request by jarkkojs**
The first one migrates to kvm_enc_region and other one fixes a compilation issues in Arch Linux.

### CODE REVIEW

**Comment by jarkkojs** [comment]
Hey, please denote me if kvm_enc_region patch still causes compilation issues. I was able to compile it locally. The only way I can test that the build server does not fail is by a PR.

File: src/backend/sev/snp/launch/mod.rs [hunk 1]
```diff
@@ -18,6 +18,11 @@ use std::os::unix::io::AsRawFd;
 
 use bitflags::bitflags;
 
+use kvm_bindings::bindings::kvm_enc_region;
+use kvm_ioctls::Kvm;
+
+use libc;
```
Comment (rvolosatovs) [inline src/backend/sev/snp/launch/mod.rs:24]: https://github.com/enarx/enarx/pull/1247#discussion_r788699935

**Review by rvolosatovs — CHANGES_REQUESTED** [review]
How about we reopen https://github.com/enarx/enarx/pull/1247 with just the SEV-specific changes (i.e. 2nd commit) and open a separate one for the build change (1st commit)?
Since you opened both PRs from `main` branch, you'd have to close this PR and push the build update to a different (new) branch
You can reopen the original PR by clicking "reopen" and then force-pushing the branch. If you have force-pushed the branch already before, then you will have to force-push it back to the latest-known hash ( `1c2ea7b` ), then reopen and force-push again to the most recent changes. This way we can keep the conversation and review comments, which are still not addressed and now scattered around between the two PRs.

File: src/backend/sev/snp/launch/mod.rs (hunk 1 above)
Comment (jarkkojs) [reply src/backend/sev/snp/launch/mod.rs]: I've now fixed this!

File: src/backend/sev/mod.rs [hunk 2]
```diff
@@ -14,8 +14,10 @@ use data::{
     dev_kvm, dev_sev, dev_sev_readable, dev_sev_writable, has_reasonable_memlock_rlimit,
     kvm_version, sev_enabled_in_kernel, CPUIDS,
 };
+
+use kvm_bindings::bindings::kvm_enc_region;
+
 use kvm_ioctls::VmFd;
```
Comment (rvolosatovs) [inline src/backend/sev/mod.rs:20]: Please follow the guidelines https://github.com/enarx/enarx/blob/main/STYLE.md#use-declarations

File: src/backend/sev/snp/launch/mod.rs [hunk 3]
```diff
@@ -18,37 +18,41 @@ use std::os::unix::io::AsRawFd;
 
 use bitflags::bitflags;
 
+use kvm_bindings::kvm_enc_region;
+
+use kvm_ioctls::VmFd;
+
```
Comment (rvolosatovs) [inline src/backend/sev/snp/launch/mod.rs:24]: IDEM

**Comment by jarkkojs** [comment]
> How about we reopen #1247 with just the SEV-specific changes (i.e. 2nd commit) and open a separate one for the build change (1st commit)? Since you opened both PRs from `main` branch, you'd have to close this PR and push the build update to a different (new) branch You can reopen the original PR by clicking "reopen" and then force-pushing the branch. If you have force-pushed the branch already before, then you will have to force-push it back to the latest-known hash ( `1c2ea7b` ), then reopen and force-push again to the most recent changes. This way we can keep the conversation and review comments, which are still not addressed and now scattered around between the two PRs.

Why that would make sense? They are separate commit with different closes tags.

File: src/backend/sev/mod.rs (hunk 2 above)
Comment (jarkkojs) [reply src/backend/sev/mod.rs:20]: I'm not sure how that does not follow the guide line. It's in its own group.

**Comment by rvolosatovs** [comment]
> > How about we reopen #1247 with just the SEV-specific changes (i.e. 2nd commit) and open a separate one for the build change (1st commit)? Since you opened both PRs from `main` branch, you'd have to close this PR and push the build update to a different (new) branch You can reopen the original PR by clicking "reopen" and then force-pushing the branch. If you have force-pushed the branch already before, then you will have to force-push it back to the latest-known hash ( `1c2ea7b` ), then reopen and force-push again to the most recent changes. This way we can keep the conversation and review comments, which are still not addressed and now scattered around between the two PRs.
> 
> Why that would make sense? They are separate commits with different closes tags. I think it'd be just extra bureaucracy without any point.

That was a response to the original state of the PR when it was identical to #1247, but with the unadressed review comments hidden (since it's a separate PR). It does not matter much at this point indeed, since the contents of the PR have since changed and most of the comments are addressed.

**Comment by jarkkojs** [comment]
> > > How about we reopen #1247 with just the SEV-specific changes (i.e. 2nd commit) and open a separate one for the build change (1st commit)? Since you opened both PRs from `main` branch, you'd have to close this PR and push the build update to a different (new) branch You can reopen the original PR by clicking "reopen" and then force-pushing the branch. If you have force-pushed the branch already before, then you will have to force-push it back to the latest-known hash ( `1c2ea7b` ), then reopen and force-push again to the most recent changes. This way we can keep the conversation and review comments, which are still not addressed and now scattered around between the two PRs.
> > 
> > 
> > Why that would make sense? They are separate commits with different closes tags. I think it'd be just extra bureaucracy without any point.
> 
> That was a response to the original state of the PR when it was identical to #1247, but with the unadressed review comments hidden (since it's a separate PR). It does not matter much at this point indeed, since the contents of the PR have since changed and most of the comments are addressed.

OK got it. Yeah, I got the point about issues and PRs but I think it'd be better idea to go forward as the changes IMHO are in good shape. You could even argue that the first change is logically a legit part of the overall patch set given that it enables compilation, which was broken before that.

File: src/backend/sev/mod.rs (hunk 2 above)
Comment (jarkkojs) [reply src/backend/sev/mod.rs:20]: So: can you just say how this is wrong and how it should be and then perhaps we can refine the documentation to make it less likely to make the same mistake?

File: src/backend/sev/snp/launch/mod.rs [hunk 4]
```diff
@@ -18,37 +18,41 @@ use std::os::unix::io::AsRawFd;
 
 use bitflags::bitflags;
 
+use kvm_bindings::kvm_enc_region;
+
+use kvm_ioctls::VmFd;
```
Comment (haraldh) [inline src/backend/sev/snp/launch/mod.rs:23]:
```suggestion
use bitflags::bitflags;
use kvm_bindings::kvm_enc_region;
use kvm_ioctls::VmFd;
```

File: src/backend/sev/mod.rs (hunk 2 above)
Comment (haraldh) [inline src/backend/sev/mod.rs:20]:
```suggestion

use kvm_bindings::bindings::kvm_enc_region;
use kvm_ioctls::VmFd;
```

**Review by haraldh — CHANGES_REQUESTED** [review]

File: src/backend/sev/mod.rs (hunk 2 above)
Comment (jarkkojs) [reply src/backend/sev/mod.rs:20]: OK, will do, thanks to everyone for patience :-)

File: src/backend/sev/snp/launch/mod.rs (hunk 3 above)
Comment (jarkkojs) [reply src/backend/sev/snp/launch/mod.rs]: OK so it should be good now :-)

File: src/backend/sev/mod.rs [hunk 5]
```diff
@@ -14,8 +14,10 @@ use data::{
     dev_kvm, dev_sev, dev_sev_readable, dev_sev_writable, has_reasonable_memlock_rlimit,
     kvm_version, sev_enabled_in_kernel, CPUIDS,
 };
+
```
Comment (rvolosatovs) [inline src/backend/sev/mod.rs:17]:
```suggestion
```

Comment (jarkkojs) [reply src/backend/sev/mod.rs:17]: I think it should be good now.

File: src/backend/sev/snp/launch/mod.rs (hunk 4 above)
Comment (jarkkojs) [reply src/backend/sev/snp/launch/mod.rs]: BTW, why there is empty line between use std::* and use::bitflags? Anyway the issue is sorted now. I'm not still exactly sure when to add a newline and when not.

File: src/backend/sev/mod.rs (hunk 5 above)
Comment (rvolosatovs) [reply src/backend/sev/mod.rs:17]: It's still not addressed.

There essentially should be only 2 groups of imports - standard library imports and external dependency imports.

**Review by npmccallum — APPROVED** [review]

**Comment by npmccallum** [comment]
@haraldh and @rvolosatovs It looks like we're just waiting for your approvals on this one.

**Review by rvolosatovs — APPROVED** [review]

**Review by haraldh — APPROVED** [review]

**Pull request merged**